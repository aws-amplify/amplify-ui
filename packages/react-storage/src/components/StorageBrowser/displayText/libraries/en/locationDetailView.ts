import { DEFAULT_LIST_VIEW_DISPLAY_TEXT } from './shared';
import { DefaultLocationDetailViewDisplayText } from '../../types';

export const DEFAULT_LOCATION_DETAIL_VIEW_DISPLAY_TEXT: DefaultLocationDetailViewDisplayText =
  {
    ...DEFAULT_LIST_VIEW_DISPLAY_TEXT,
    getListResultsMessage: () => 'help me',
    searchExhaustedMessage: 'Showing results for up to the first 10,000 items',
    searchIncludeSubfoldersLabel: 'Include subfolders',
    searchPlaceholder: 'Search current folder',
    tableColumnLastModifiedHeader: 'Last modified',
    tableColumnNameHeader: 'Name',
    tableColumnSizeHeader: 'Size',
    tableColumnTypeHeader: 'Type',
    title: (location) => {
      const { current, key } = location;
      const { bucket = '', prefix } = current ?? {};
      return prefix ? key : bucket;
    },
  };
