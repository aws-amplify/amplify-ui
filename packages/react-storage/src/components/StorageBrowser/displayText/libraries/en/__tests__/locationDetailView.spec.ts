import { LIST_ITEMS_SCENARIOS } from './scenarios';
import { DEFAULT_LOCATION_DETAIL_VIEW_DISPLAY_TEXT } from '../locationDetailView';

describe('LocationDetailView display text', () => {
  it('should match snapshot values', () => {
    expect(DEFAULT_LOCATION_DETAIL_VIEW_DISPLAY_TEXT).toMatchSnapshot();
  });

  it.each(LIST_ITEMS_SCENARIOS)(
    '`getListResultsMessage` returns the expected values in the %s scenario',
    (_, data) => {
      const { getListItemsResultMessage } =
        DEFAULT_LOCATION_DETAIL_VIEW_DISPLAY_TEXT;

      expect(getListItemsResultMessage(data)).toMatchSnapshot();
    }
  );

  it('returns correct string for title', () => {
    const { getTitle } = DEFAULT_LOCATION_DETAIL_VIEW_DISPLAY_TEXT;

    expect(
      getTitle({
        current: {
          bucket: 'test-bucket',
          permissions: ['get', 'list'],
          id: '123',
          prefix: '',
          type: 'PREFIX',
        },
        path: '',
        key: 'path/to/somewhere',
      })
    ).toBe(`path/to/somewhere`);

    expect(
      getTitle({
        current: {
          bucket: 'test-bucket',
          permissions: ['get', 'list'],
          id: '123',
          prefix: '',
          type: 'PREFIX',
        },
        path: '',
        key: '',
      })
    ).toBe(`test-bucket`);
  });
});
