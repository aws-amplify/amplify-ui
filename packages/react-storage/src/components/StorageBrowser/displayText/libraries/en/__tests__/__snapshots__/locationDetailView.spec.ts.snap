// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetailView display text \`getListResultsMessage\` returns the expected values in the empty results scenario 1`] = `
{
  "content": "No files.",
  "type": "info",
}
`;

exports[`LocationDetailView display text \`getListResultsMessage\` returns the expected values in the empty search results scenario 1`] = `
{
  "content": "No files.",
  "type": "info",
}
`;

exports[`LocationDetailView display text \`getListResultsMessage\` returns the expected values in the failed scenario 1`] = `
{
  "content": "Network got confused",
  "type": "error",
}
`;

exports[`LocationDetailView display text \`getListResultsMessage\` returns the expected values in the loading scenario 1`] = `undefined`;

exports[`LocationDetailView display text \`getListResultsMessage\` returns the expected values in the search exhausted scenario 1`] = `
{
  "content": "Showing results for up to the first 10,000 items.",
  "type": "info",
}
`;

exports[`LocationDetailView display text \`getListResultsMessage\` returns the expected values in the search failed scenario 1`] = `
{
  "content": "Network got confused",
  "type": "error",
}
`;

exports[`LocationDetailView display text \`getListResultsMessage\` returns the expected values in the search limit reached scenario 1`] = `undefined`;

exports[`LocationDetailView display text should match snapshot values 1`] = `
{
  "getActionListItemLabel": [Function],
  "getDateDisplayValue": [Function],
  "getListItemsResultMessage": [Function],
  "getTitle": [Function],
  "loadingIndicatorLabel": "Loading",
  "searchClearLabel": "Clear search",
  "searchPlaceholder": "Search current folder",
  "searchSubfoldersToggleLabel": "Include subfolders",
  "searchSubmitLabel": "Submit",
  "selectAllFilesLabel": "Select all files",
  "selectFileLabel": "Select file",
  "tableColumnLastModifiedHeader": "Last modified",
  "tableColumnNameHeader": "Name",
  "tableColumnSizeHeader": "Size",
  "tableColumnTypeHeader": "Type",
}
`;
